src/App.tsx

```diff
@@ -14,11 +15,89 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedProducts: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedProducts: ['Jam'],
+  };
+
+  modifySelectedProducts = (good: string) => {
+    const { selectedProducts } = this.state;
+
+    this.setState({
+      selectedProducts: (
+        selectedProducts.includes(good)
+          ? selectedProducts.filter(item => item !== good)
+          : [...selectedProducts, good]
+      ),
+    });
+  };
+
+  clearAll = () => {
+    this.setState({ selectedProducts: [] });
+  };
+
+  render() {
+    const { selectedProducts } = this.state;
+
+    return (
+      <div className="App">
+        <h1
+          className="basket"
+        >
+          {selectedProducts.length === 0 ? 'No goods selected'
+            : (selectedProducts.length === 1 ? `${selectedProducts} is selected`
+              : `${selectedProducts.join(', ')} are selected`)}
+
+          <button
+            type="button"
+            className={
+              `basket__button
+              button
+              ${selectedProducts.length === 0
+        ? 'basket__button--empty'
+        : 'basket__button--not-empty'}`
+            }
+            onClick={() => this.clearAll()}
+          >
+            Clear
+          </button>
+        </h1>
+
+        <div className="goods">
+          {goodsFromServer.map(good => (
+            <button
+              type="button"
+              className={
+                `good
+                ${!selectedProducts.includes(good)
+              ? 'good--not-selected'
+              : 'good--selected'}`
+              }
+            >
+              {good}
+              <button
+                type="button"
+                className={
+                  `good__button
+                  button
+                  ${!selectedProducts.includes(good)
+              ? 'good__button--not-selected'
+              : 'good__button--selected'}`
+                }
+                onClick={() => this.modifySelectedProducts(good)}
+              >
+                {selectedProducts.includes(good) ? 'Remove' : 'Select'}
```

Comment:
`selectedProducts.includes(good)` - you use this several times, so makes sence to create a variable for this